plat/cavium/common/bl31_thunder_setup.c

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright (c) 2013-2014, ARM Limited and Contributors. All rights reserved.
+ * Copyright (c) 2016-2017, Cavium Inc. All rights reserved.<BR>
+ * Copyright (c) 2018     , Facebook, Inc.
```

Comment:
Are you authorised to contribute this code on behalf of Facebook (and Cavium)? If so, please use only the first copyright notice.

Reply:
Yes.
Though changing the copyright seems tricky.
IANAL but it seems wrong, even when it's released under BSD license.

From BSD license:
"a. Redistributions of source code must retain the above copyright notice,"

Reply:
If you are authorised to contribute this code on behalf of Facebook and Cavium then you can ask their permission to change the copyright notice. It would not change who the underlying copyright holders are. But it sounds like you don't have such authorisation from earlier comments.

Reply:
Hello, I'm an engineer from Facebook working with 9elements and Cavium. From our standpoint, yes, 9elements has authorization to release this code. Since 9elements is working for Facebook on contract, the copyright change made here is appropriate AFAICT.

Some additional background: The sources here are derived from Cavium's Octeon-TX SDK: https://github.com/Cavium-Open-Source-Distributions/OCTEON-TX-SDK

Basically you'll find a ~1 year old copy of ATF in there, which we've forward ported to upstream. 9elements has driven mostly driven this effort and has tested it.